lib/signaling/v2/peerconnection.js

```diff
@@ -363,7 +370,15 @@ PeerConnectionV2.prototype._handleSignalingStateChange = function _handleSignali
  */
 PeerConnectionV2.prototype._handleTrackEvent = function _handleTrackEvent(event) {
   var mediaStreamTrack = event.track;
-  this.emit('trackAdded', mediaStreamTrack);
+  var mediaTrackReceiver = new MediaTrackReceiver(mediaStreamTrack.id, mediaStreamTrack);
+  this._mediaTrackReceivers.add(mediaTrackReceiver);
+
+  var self = this;
+  mediaStreamTrack.addEventListener('ended', function() {
+    self._mediaTrackReceivers.delete(mediaTrackReceiver);
+  });
```

Comment:
This will not fire if `stop` is called. Perhaps this is OK with our usage here?

Reply:
@markandrus I though this should fire at least in Chrome? We don't really remove the tracks in FF/Safari.

Reply:
Per [mediacapture-main](https://w3c.github.io/mediacapture-main/),

> When a MediaStreamTrack track ends for any reason other than the stop() method being invoked, the User Agent must queue a task that runs the following steps:
>
> ...
>
> 4. Fire a simple event named ended at the object.

"ended" should not fire if `stop` is called. It's OK to have this if we are clear about the scenario we are using it for; however, it may not be necessary.

Reply:
@markandrus So when is "ended" fired on a remote MediaStreamTrack?